src/components/Input/Input.js

```diff
@@ -14,42 +15,48 @@ class Input extends PureComponent {
       style: PropTypes.object,
       theme: PropTypes.object,
       type: PropTypes.string,
-      value: PropTypes.string
+      value: PropTypes.string,
     };
   }
 
   static get defaultProps() {
     return {
       onChange: () => {},
-      type: 'text'
+      type: 'text',
     };
   }
 
   render() {
     const {
-      type,
       children,
       name,
       onChange,
       placeholder,
-      theme,
       style,
+      theme,
+      type,
       value,
       ...otherProps
     } = this.props;
 
-    return (
-      <input
-        className={theme.input[type]}
-        name={name}
-        onChange={onChange}
-        placeholder={placeholder}
-        style={style}
-        type={type}
-        value={value}
-        {...otherProps}
-      />
-    );
+    switch (type) {
+      case 'radio':
+        return <Radio {...this.props}>{children}</Radio>;
```

Comment:
Nice! I like this a lot. Also adds a lot more flexibility for more custom input types in the future 👍